File: Programmers/java/lev0/분수의 덧셈/solution.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = {(numer1 *denom2 + numer2*denom1) , (denom1 * denom2) } ;
        int num = answer[0]>answer[1]?answer[0]:answer[1];
        for (int i =(int)num/2 ; i>1;i--)
            if (answer[0]%i ==0 && answer[1]%i ==0)
            {
                answer [0] = (int)answer[0]/i;
                answer[1] = (int)answer[1]/i;
            }
                
        return answer;
        
    }
}

File: Programmers/py/lev0/분수의 덧셈/solution.py
def solution(numer1, denom1, numer2, denom2):
    parent = (denom1 * denom2)
    child = (numer1*denom2 + denom1 * numer2) 
    num = parent if parent>child else child 
    
    for i in range(num,int(num/2)+1,-1):
        if parent%i == 0 and child %i==0:
            parent = int(parent/i)
            child = int(child/i)
    return child , parent  

print(solution(1,2,3,4))